src/core/lib/security/authorization/rbac_translator.cc

```diff
@@ -0,0 +1,351 @@
+}
+
+absl::StatusOr<std::map<std::string, Rbac::Policy>> ParseRulesArray(
+    const Json& json, absl::string_view name) {
+  std::map<std::string, Rbac::Policy> policies;
+  for (size_t i = 0; i < json.array_value().size(); ++i) {
+    const Json& child = json.array_value().at(i);
+    if (child.type() != Json::Type::OBJECT) {
+      return absl::InvalidArgumentError(
+          absl::StrCat("rules ", i, ": is not an object."));
+    }
+    auto it = child.object_value().find("name");
+    if (it == child.object_value().end()) {
+      return absl::InvalidArgumentError(
+          absl::StrCat("rules ", i, ": \"name\" is not present."));
+    }
+    if (it->second.type() != Json::Type::STRING) {
+      return absl::InvalidArgumentError(
+          absl::StrCat("rules ", i, ": \"name\" is not a string."));
+    }
+    std::string policy_name =
+        std::string(name) + "_" + it->second.string_value();
+    auto policy_or = ParseRules(child);
+    if (!policy_or.ok()) {
+      return absl::Status(
+          policy_or.status().code(),
+          absl::StrCat("rules ", i, ": ", policy_or.status().message()));
+    }
+    policies[policy_name] = std::move(policy_or.value());
+  }
+  return std::move(policies);
+}
+
+absl::StatusOr<Rbac> ParseDenyRulesArray(const Json& json,
+                                         absl::string_view name) {
+  auto policies_or = ParseRulesArray(json, name);
+  if (!policies_or.ok()) return policies_or.status();
+  return Rbac(Rbac::Action::DENY, std::move(policies_or.value()));
+}
+
+absl::StatusOr<Rbac> ParseAllowRulesArray(const Json& json,
+                                          absl::string_view name) {
+  auto policies_or = ParseRulesArray(json, name);
+  if (!policies_or.ok()) return policies_or.status();
+  return Rbac(Rbac::Action::ALLOW, std::move(policies_or.value()));
+}
+
+}  // namespace
+
+absl::StatusOr<RbacPolicies> GenerateRbacPolicies(
+    absl::string_view authz_policy) {
+  grpc_error* error = GRPC_ERROR_NONE;
+  Json json = Json::Parse(authz_policy, &error);
+  if (error != GRPC_ERROR_NONE) {
+    absl::Status status = absl::InvalidArgumentError(
+        absl::StrCat("Failed to parse SDK authorization policy. Error: ",
+                     grpc_error_string(error)));
+    GRPC_ERROR_UNREF(error);
+    return status;
+  }
+  if (json.type() != Json::Type::OBJECT) {
+    return absl::InvalidArgumentError(
+        "SDK authorization policy is not an object.");
+  }
+  auto it = json.mutable_object()->find("name");
+  if (it == json.mutable_object()->end()) {
+    return absl::InvalidArgumentError("\"name\" field is not present.");
+  }
+  if (it->second.type() != Json::Type::STRING) {
+    return absl::InvalidArgumentError("\"name\" is not a string.");
+  }
+  absl::string_view name = it->second.string_value();
+  RbacPolicies rbac_policies;
+  it = json.mutable_object()->find("deny_rules");
+  if (it != json.mutable_object()->end()) {
```

Comment:
When there is no deny_rules, the `rbac_policies.deny_policy` is empty. In RBAC engine, the default enum is ALLOW. We may end up have deny_policy set ALLOW action by mistake (as default value). We shall explicitly set `rbac_policies.deny_policy` action to DENY to be self-contain. 

Please also update unit test accordingly.